programs/pluto/ikev1_quick.c

```diff
@@ -602,7 +602,12 @@ check_net_id(struct isakmp_ipsec_id *id
     } else if(!end->has_client && !subnetisaddr(&net_temp, endip)) {
         loglog(RC_LOG_SERIOUS, "%s subnet returned does not match my self-proposal - us:%s vs them:%s",
                which,subxmt,subrec);
-        bad_proposal = TRUE;
+#ifdef ALLOW_MICROSOFT_BAD_PROPOSAL
```

Comment:
Since both sides of this if()/else if(), result in the same exception being applied, the overall affect of ALLOW_MICROSOFT_BAD_PROPOSAL is to do nothing, and log the same thing, please just surround the entire if/else of an #ifdef, and don't even do the test.  
Please document what versions of which Microsoft products do this.
This should be set on a per-conn basis, I think.  Or perhaps, enabled by a vendor-based impairment.
I would have removed this #ifdef had I noticed it before.

Reply:
There's a subtle difference between the messages in the if/elseif branches, which clearly highlights where the problem is. Grouping them together would make that harder, so if anything it's better to keep it as-is.

The device is Ubiquiti/UniFi (don't know further details sorry), plus a search through the webs shows that many other vendors need this. So if anything the workaround name is misleading. Considering the workaround has been the default for years, removing it sounds like a terrible move.

Reply:
Humble poke?